x/estaking/keeper/query_edenb_burn.go

```diff
@@ -0,0 +1,71 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/elys-network/elys/x/estaking/types"
+	ptypes "github.com/elys-network/elys/x/parameter/types"
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) EdenBBurnAmount(goCtx context.Context, req *types.QueryEdenBBurnAmountRequest) (*types.QueryEdenBBurnAmountResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	delegator, err := sdk.AccAddressFromBech32(req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get commitments
+	commitments := k.commKeeper.GetCommitments(ctx, delegator)
+
+	// Get previous amount
+	prevElysStaked := k.GetElysStaked(ctx, delegator)
+	if prevElysStaked.Amount.IsZero() {
+		return nil, status.Error(codes.NotFound, "no staked Elys")
+	}
+
+	edenCommitted := commitments.GetCommittedAmountForDenom(ptypes.Eden)
+
+	// Total EdenB amount
+	edenBCommitted := commitments.GetCommittedAmountForDenom(ptypes.EdenB)
+	edenBClaimed := commitments.GetClaimedForDenom(ptypes.EdenB)
+	totalEdenB := edenBCommitted.Add(edenBClaimed)
+
+	edenBToBurn := osmomath.ZeroBigDec()
+
+	if req.TokenType == types.TokenType_TOKEN_TYPE_ELYS {
+		// Unstaked
+		unstakedElysDec := osmomath.BigDecFromSDKInt(req.Amount)
+		edenCommittedAndElysStakedDec := osmomath.BigDecFromSDKInt(edenCommitted.Add(prevElysStaked.Amount))
+		if edenCommittedAndElysStakedDec.GT(osmomath.ZeroBigDec()) {
+			edenBToBurn = unstakedElysDec.Quo(edenCommittedAndElysStakedDec).Mul(osmomath.BigDecFromSDKInt(totalEdenB))
```

Comment:
What if unstakedElysDec > edenCommittedAndElysStaked, should we also add a check for this, to update unstakedElysDec = edenCommittedAndElysStaked

Reply:
that would be wrong scenario as total staked can't be less than a single stake by any user

Reply:
added additional check to reject this kind of query